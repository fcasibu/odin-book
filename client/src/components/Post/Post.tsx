import Card from "../Card";
import Form, { TextArea } from "../Form";

type PostProps = {
    children: React.ReactNode;
};

type PostBodyProps = {
    postImage?: string;
};

export const PostHeader = () => {
    return (
        <div className="flex justify-between">
            <div className="flex gap-2">
                <div className="rounded-full max-w-[40px] overflow-hidden cursor-pointer">
                    <img
                        src="https://avatars.githubusercontent.com/u/75290989?s=400&v=4"
                        className="object-cover"
                    />
                </div>
                <div className="flex flex-col cursor-pointer">
                    <span className="font-semibold text-sm">John Doe</span>
                    <span className="text-slate-500 text-xs">September 18 at 4:44 PM</span>
                </div>
            </div>
            <button className="font-bold text-md self-start">...</button>
        </div>
    );
};

export const PostBody = ({ postImage }: PostBodyProps) => {
    return (
        <>
            <div>
                <p className="text-sm">
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
                    incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis
                    nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
                    Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu
                    fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
                    culpa qui officia deserunt mollit anim id est laborum.
                </p>
            </div>
            {postImage && (
                <div className="mx-[-16px] cursor-pointer">
                    <img src={postImage} alt="" className="object-cover w-full" />
                </div>
            )}
            <div className="flex justify-between items-center text-sm text-slate-500">
                <span className="cursor-pointer">Carl, Mary, and 18 others</span>
                <span className="cursor-pointer">3 Comment</span>
            </div>
            <div className="flex justify-center items-center border-y-2 border-gray-100">
                <button className="text-sm w-full rounded-md p-2 hover:bg-slate-100 active:bg-slate-200">
                    Like
                </button>
                <button className="text-sm w-full rounded-md p-2 hover:bg-slate-100 active:bg-slate-200">
                    Comment
                </button>
            </div>
        </>
    );
};

export const PostCommentBox = () => {
    return (
        <div className="flex gap-2">
            <div className="rounded-full h-fit w-[40px] overflow-hidden cursor-pointer">
                <img
                    src="https://avatars.githubusercontent.com/u/75290989?s=400&v=4"
                    className="object-cover"
                />
            </div>
            <Form>
                <TextArea id="comment" name="comment" />
            </Form>
        </div>
    );
};

export const Post = ({ children }: PostProps) => {
    return (
        <Card className="flex flex-col gap-2">
            <PostHeader />
            <PostBody postImage="https://picsum.photos/536/354" />
            {children}
            <PostCommentBox />
        </Card>
    );
};
