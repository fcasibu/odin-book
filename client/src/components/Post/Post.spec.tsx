import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Post } from "./Post";

describe("Post", () => {
    it("should render and match the snapshot", () => {
        const { container } = render(<Post> </Post>);

        expect(container).toMatchSnapshot();
    });
})

