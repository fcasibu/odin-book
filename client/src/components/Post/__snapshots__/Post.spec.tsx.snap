// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should render and match the snapshot 1`] = `
<div>
  <div
    class="bg-white rounded-md p-4 shadow-md flex flex-col gap-2"
  >
    <div
      class="flex justify-between"
    >
      <div
        class="flex gap-2"
      >
        <div
          class="rounded-full max-w-[40px] overflow-hidden cursor-pointer"
        >
          <img
            class="object-cover"
            src="https://avatars.githubusercontent.com/u/75290989?s=400&v=4"
          />
        </div>
        <div
          class="flex flex-col cursor-pointer"
        >
          <span
            class="font-semibold text-sm"
          >
            John Doe
          </span>
          <span
            class="text-slate-500 text-xs"
          >
            September 18 at 4:44 PM
          </span>
        </div>
      </div>
      <button
        class="font-bold text-md self-start"
      >
        ...
      </button>
    </div>
    <div>
      <p
        class="text-sm"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </p>
    </div>
    <div
      class="mx-[-16px] cursor-pointer"
    >
      <img
        alt=""
        class="object-cover w-full"
        src="https://picsum.photos/536/354"
      />
    </div>
    <div
      class="flex justify-between items-center text-sm text-slate-500"
    >
      <span
        class="cursor-pointer"
      >
        Carl, Mary, and 18 others
      </span>
      <span
        class="cursor-pointer"
      >
        3 Comment
      </span>
    </div>
    <div
      class="flex justify-center items-center border-y-2 border-gray-100"
    >
      <button
        class="text-sm w-full rounded-md p-2 hover:bg-slate-100 active:bg-slate-200"
      >
        Like
      </button>
      <button
        class="text-sm w-full rounded-md p-2 hover:bg-slate-100 active:bg-slate-200"
      >
        Comment
      </button>
    </div>
     
    <div
      class="flex gap-2"
    >
      <div
        class="rounded-full h-fit w-[40px] overflow-hidden cursor-pointer"
      >
        <img
          class="object-cover"
          src="https://avatars.githubusercontent.com/u/75290989?s=400&v=4"
        />
      </div>
      <form
        class="flex flex-col gap-2 w-full"
      >
        <label
          for="comment"
        >
          <textarea
            class="w-full rounded-sm p-2 border-2 focus:outline-sky-500 resize-none no-scrollbar"
            id="comment"
            name="comment"
          />
        </label>
      </form>
    </div>
  </div>
</div>
`;
